Teleport Identity Security - Add Identity Activity Center alerts documentation

benarent opened the pull request:
Introduces a new documentation page describing pre-built security detections and alerting features in the Identity Activity Center. Includes an overview of supported integrations (AWS, GitHub, Okta, Teleport), detection categories, severity levels, and planned alert management features. 

Followup for https://github.com/gravitational/teleport/pull/58939#discussion_r2388748605

benarent committed:
Add Identity Activity Center alerts documentation

Introduces a new documentation page describing pre-built security detections and alerting features in the Identity Activity Center. Includes an overview of supported integrations (AWS, GitHub, Okta, Teleport), detection categories, severity levels, and planned alert management features. Adds supporting image for visual reference.

benarent committed:
Update alerts.mdx

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Account compromises
+- Policy violations
+
+All detections are pre-configured and currently cannot be modified. The severity levels (Critical, High, Medium, Low) are defined by the Identity Security Team based on the potential impact of each detection.
```
```suggestion
All detections are pre-configured and currently cannot be modified. The severity levels (Critical, High, Medium, Low) are defined by the Identity Security team based on the potential impact of each detection.
```
"team" is a common noun here

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+![Alerts](../../../img/access-graph/alerts.png)
+
+
+## Detection Categories
```
```suggestion
## Detection categories
```
We use sentence case in headings.

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+## Detection Categories
+
+### AWS Detections
```
```suggestion
### AWS detections
```
Sentence case in headings

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+### AWS Detections
+
+Identity Activity Center monitors your AWS infrastructure for critical security events:
```
```suggestion
Teleport Identity Security monitors your AWS infrastructure for critical security events:
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+Identity Activity Center monitors your AWS infrastructure for critical security events:
+
+**Critical & High Severity:**
```
```suggestion
**Critical and high severity:**
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Exposed credential policy modifications
+- Login profile updates
+
+**Medium & Low Severity:**
```
```suggestion
**Medium and low severity:**
```
Since this is regular prose, rather than a product name, I would use "and" instead of an ampersand for consistency.

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- DB snapshot attribute modifications
+- AWS key deletion or disabling
+
+### GitHub Detections
```
```suggestion
### GitHub detections
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+Monitor your GitHub organizations and repositories for security-relevant changes:
+
+**Critical & High Severity:**
```
```suggestion
**Critical and high severity:**
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Secret scanning alerts
+- Organization member updates
+
+**Medium & Low Severity:**
```
```suggestion
**Medium and low severity:**
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Organization moderator additions
+- Integration installations
+
+**Advanced Security Detections:**
```
The bold text is going to render in the same paragraph as the text below it. Is that the intention? If we want this to be a heading, we should start it with a `####` for semantic HTML.

benarent replied:
Ty. Updated this

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+**Advanced Security Detections:**
+The system includes 26 sub-types of GitHub advanced security change detections that monitor for modifications to security features like Dependabot alerts.
+
+### Okta Detections
```
```suggestion
### Okta detections
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+Track identity and access management events in your Okta environment:
+
+**High Severity:**
```
```suggestion
**High severity:**
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- API token creation/revocation
+- Security threat events
+
+**Medium Severity:**
```
```suggestion
**Medium severity:**
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Dormant account access
+- Support-initiated password or MFA resets
+
+### Teleport Detections
```
```suggestion
### Teleport detections
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+Monitor Teleport-specific security events:
+
+**High Severity:**
```
```suggestion
**High severity:**
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Connector updates
+- Unusual session commands
+
+### Cross-Platform Detections
```
```suggestion
### Cross-platform detections
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+### Cross-Platform Detections
+
+**Impossible Travel Detection** (GitHub, Okta, Teleport):
```
```suggestion
**Impossible travel detection (GitHub, Okta, Teleport):**
```
For consistency with other bold text used as a list heading

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Severity: High
+- Triggers when login attempts occur from multiple geographic locations within a timeframe that makes legitimate travel physically impossible
+
+## Alert Management
```
```suggestion
## Alert management
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Triggers when login attempts occur from multiple geographic locations within a timeframe that makes legitimate travel physically impossible
+
+## Alert Management
+
```
What does alert management mean in this context? I would add a short paragraph explaining what this section is about.

benarent replied:
removed this section

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+
+## Alert Management
+
+### Current Capabilities
+- View alerts across all integrated systems
+- Filter by severity level
+- Investigate related events
```
I would expand each of these into a sentence. Otherwise, I'm not sure what each item means in terms of what I can do with the Web UI.

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Filter by severity level
+- Investigate related events
+
+### Planned Features
+- Alert resolution workflows
+- Acknowledgment and muting capabilities
+- Forwarding alerts to external services 
+- Custom detection creation
```
We don't typically publish roadmap items in the docs outside of the Upcoming Releases page. Would it make sense to add this there or remove it?

benarent replied:
I've removed this

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+- Forwarding alerts to external services 
+- Custom detection creation
+
+## Frequently Asked Questions
```
```suggestion
## Frequently asked questions
```

ptgott commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,140 @@
+For new detection requests based on customer needs, reach out to the Identity Security Team. Customer feedback is welcomed and helps prioritize new detection development.
+
+
+## Next Steps
```
```suggestion
## Next steps
```

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Apply suggestion from @ptgott

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Update docs/pages/identity-security/usage/alerts.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Update docs/pages/identity-security/usage/alerts.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Update docs/pages/identity-security/usage/alerts.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Update docs/pages/identity-security/usage/alerts.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Update docs/pages/identity-security/usage/alerts.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Update docs/pages/identity-security/usage/alerts.mdx

Co-authored-by: Paul Gottschling <paul.gottschling@goteleport.com>

benarent committed:
Merge branch 'benarent/docs/alerts-page' of https://github.com/gravitational/teleport into benarent/docs/alerts-page

benarent committed:
remove planned features for now

benarent committed:
Update alerts.mdx

benarent committed:
remove tags

benarent committed:
PR Feedback

benarent committed:
Fixes https://github.com/gravitational/teleport/pull/59714#discussion_r2389052588

benarent committed:
Update alerts.mdx

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+ - identity-security
+---
+
+Teleport Identity Security provides 50 pre-built security detections that automatically create alerts for suspicious identity-related activities across your infrastructure. These detections monitor events from Teleport and integrated services like AWS, GitHub, and Okta to identify potential security risks.
```
Do we need to call out the specific number of detections? I don't think knowing the exact number adds any value, and it is likely to go out of date as we refine them in the future.

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+![Alerts](../../../img/access-graph/alerts.png)
+
+
+## Detection categories
```
```suggestion
## Detection categories

```

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+#### Advanced Security Detections
+The system includes 26 sub-types of GitHub advanced security change detections that monitor for modifications to security features like Dependabot alerts.
+
+Set up GitHub detections: [GitHub Settings](../integrations/github.mdx)
```
This feels like an incomplete thought. Is this a prerequisite for the GitHub Advanced Security detections, or for all GitHub detections?

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+
+### AWS detections
+
+Teleport Identity Security monitors your AWS infrastructure for critical security events:
```
- I would end the sentence with a `.`, not a `:`
- Why do we only mention "critical security events" here, but then enumerate a bunch of other (non-critical) detections below?

benarent replied:
Clarified

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+- DB snapshot attribute modifications
+- AWS key deletion or disabling
+
+Set up AWS detections: [AWS Settings](../integrations/aws-sync.mdx)
```
The placement of this link makes it read as if this setup is only required for the medium and low severity detections. Recommend moving it up to the intro section.

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+### Cross-platform detections
+
+#### Impossible travel detection (GitHub, Okta, Teleport)
+- Severity: High
```
We should be consistent with how we indicate severity.

In previous sections, we had a separate section header for critical, high, medium, etc.

In this section we're listing the severity as a bullet point.

benarent replied:
Good feedback, fixed in a commit

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+This feature is planned for future release.
+
+### How can I request new detection types?
+For new detection requests based on customer needs, reach out to the Identity Security Team. Customer feedback is welcomed and helps prioritize new detection development.
```
How do we expect the reader to know how to get in touch with the Identity Security team?

zmb3 commented on docs/pages/identity-security/usage/alerts.mdx:
```diff
@@ -0,0 +1,139 @@
+
+## Next steps
+
+- Navigate to **Identity Security** > **Alerts** in the Teleport Web UI to view active alerts
```
This feels like something that could be mentioned at the very beginning of the document ("Identity Activity Center alerts can be accessed from the ...") instead of as a next step (which is typically links to further reading or related topics)

benarent committed:
fixes https://github.com/gravitational/teleport/pull/59714#discussion_r2392647975

benarent committed:
Removes hard coded detection number per https://github.com/gravitational/teleport/pull/59714#discussion_r2392646161

benarent committed:
AWS Detections now match list below https://github.com/gravitational/teleport/pull/59714#discussion_r2392654610

benarent committed:
moved location of link https://github.com/gravitational/teleport/pull/59714#discussion_r2392656371

benarent committed:
keeps cross platform issues consistent with rest of issues

https://github.com/gravitational/teleport/pull/59714#discussion_r2392659109

benarent committed:
added a link to get in touch with us

benarent committed:
Move view alerts to the top vs next steps

benarent committed:
Simplifed the messaging and added a 'tip' that this will be coming soon to Teleport Enterprise Cloud

Resolves https://github.com/gravitational/teleport/pull/59714#discussion_r2389019318
